### ISSUE TITLE
kubectl continues with operation even if context specified by `--context` does not exist.

### ISSUE THREAD

**Issue by mortenlj** [kubernetes/kubernetes#38864]
**Is this a request for help?**
(If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

No

**What keywords did you search in Kubernetes issues before filing this one?**
(If you have found any duplicates, you should instead reply there.):

kubectl context

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Kubernetes version** (use `kubectl version`):
╼ kubectl version
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.4", GitCommit:"3b417cc4ccd1b8f38ff9ec96bb50a81ca0ea9d56", GitTreeState:"clean", BuildDate:"2016-10-21T02:48:38Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Cloud provider or hardware configuration**:
N/A
- **OS** (e.g. from /etc/os-release):
Funtoo Linux
- **Kernel** (e.g. `uname -a`):
Linux fimojoha-w530 4.4.21-gentoo #1 SMP Mon Oct 17 14:05:24 CEST 2016 x86_64 Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz GenuineIntel GNU/Linux
- **Install tools**: N/A
- **Others**: N/A

**What happened**:
When using the `--context` parameter to `kubectl` with an incorrectly spelled context, it attempts to perform the operation usin an api-server at `http://localhost:8080`, which may or may not respond.

If an api-server was listening on that port, it could potentially have disastrous effects.

**What you expected to happen**:
I expected to get an error, and failing to execute the command.

**How to reproduce it** (as minimally and precisely as possible):

    # kubectl --context=does-not-exist version
    The connection to the server localhost:8080 was refused - did you specify the right host or port?

**Anything else do we need to know**:
Even if you turn logging up to max, there is no hint at all about why kubectl suddenly wants to use `http://localhost:8080` instead of the api-server you wanted to use, so if you have a tiny spelling error, tracking down the problem could be incredibly frustrating.

**Comment by xingzhou** [comment]
Did not see the issue by using the latest code, when issueing `kubectl --context=does-not-exist version`, I got the response of:
```
Client Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-alpha.1.9188+cfe1599983f971-dirty", GitCommit:"cfe1599983f971f1d3293cb20dce571c23cbbec2", GitTreeState:"dirty", BuildDate:"2016-12-19T02:53:59Z", GoVersion:"go1.6", Compiler:"gc", Platform:"linux/amd64"}
error: context "does-not-exist" does not exist
```
Noticed that in the last line, the output mentioned that the context does not exist.

**Comment by mortenlj** [comment]
I've tested with the version of kubectl that comes with the Google Cloud SDK, v1.4.6, and the issue still persists. However, when I downloaded the 1.5.1 version, it's gone, so I guess this problem has been resolved somewhere between 1.4.6 and current.

Closing the issue.
